functions/src/scheduler/schedule.ts

```diff
@@ -23,3 +25,29 @@ export const deletePingsAfter30days: any = functions.pubsub.schedule('00 00 * *
     });
     return Promise.all(promises);
   });
+
+export const runAllMonitors60Mins: any = functions.pubsub.schedule('every 60 mins')
+  .onRun(async () => {
+    const snapshots: QuerySnapshot = await FirebaseAdmin.firestore()
+      .collection('projects')
+      .get();
+    
+    snapshots.docs.forEach(async (doc: firestore.QueryDocumentSnapshot) => {
+      const project: ProjectModel = <ProjectModel>doc.data();
+      try {
+        if (project.url && Array.isArray(project.monitors) && project.monitors.length > 0) {
+          const promises: Promise<WriteResult>[] = [];
+
+          for (const monitor of project.monitors) {
+            promises.push(ping(project.uid, monitor.uid))
+          }
+          await Promise.all(promises);
+        }
+        await Promise.all([]);
```

Comment:
I think instead of creating a new `Promise.all([])`, we should return the `Promise.all(promises)` which could be an empty array

Reply:
My thoughts:

- define the `promises` outside the `try` 
- return `Promise.all(promises)` after the `try/catch`

Reply:
Sure!!